dms/models/dms_access_groups.py

```diff
@@ -148,22 +174,21 @@ def default_get(self, fields_list):
     )
     def _compute_users(self):
         for record in self:
-            users = record.mapped("group_ids.users")
-            users |= record.mapped("explicit_user_ids")
-            users |= record.mapped("parent_group_id.users")
+            users = (
+                record.group_ids.users
+                | record.explicit_user_ids
+                | record.parent_group_id.users
+            )
             record.update({"users": users, "count_users": len(users)})
 
-    @api.constrains("parent_path")
-    def _check_parent_recursiveness(self):
-        """Forbid recursive relationships."""
-        for one in self:
-            if not one.parent_group_id:
-                continue
-            if str(one.id) in one.parent_path.split("/"):
-                raise ValidationError(
-                    _("Parent group '%(parent)s' is child of '%(current)s'.")
-                    % {
-                        "parent": one.parent_group_id.display_name,
-                        "current": one.display_name,
-                    }
-                )
+    # ----------------------------------------------------------
+    # CRUD/ORM Overrides
+    # ----------------------------------------------------------
+    def copy(self, default=None):
+        default = dict(default or {})
+        default["name"] = _("%s (copy)") % self.name
+        return super().copy(default=default)
+
+    def write(self, vals):
+        self.env.registry.clear_cache()
```

Comment:
Why this? When you do weird things, you must always put a comment saying why. If not, any other review/migration, this will be removed if we don't find the reason why it's put.

Reply:
That's some old code I might have written. I'm removing it.